resources/js/beatmapsets-show/controller.ts

```diff
@@ -70,6 +82,51 @@ export default class Controller {
     return this.beatmaps.get(this.currentBeatmap.mode) ?? [];
   }
 
+  get relatedTags() {
+    const map = new Map();
+
+    for (const tag of this.beatmapset.related_tags) {
+      map.set(tag.id, tag);
+    }
+
+    return map;
+  }
+
+  @computed
+  get tags() {
+    const mapperTagSet = new Set(this.beatmapset.tags.split(' ').filter(present));
```

Comment:
I don't know about deduplicating the old tag words. It may "break" titles/names like repeated `no` and other stuff which is why it was not done in the first place.